android/src/main/java/com/reactnativefileutils/FileUtilsModule.java

```diff
@@ -1,34 +1,170 @@
+    // contents as the file is likely a video.
+    if (extension != null && !extension.isEmpty()) {
+      type = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension);
+      promise.resolve(type);
+      return;
     }
 
+    try {
+      Uri fileUri = Uri.parse(uri);
+      MediaMetadataRetriever retriever = GetMediaMetadataRetriever(fileUri);
+      String mimeType = retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_MIMETYPE);
+      retriever.release();
 
-    // Example method
-    // See https://reactnative.dev/docs/native-modules-android
-    @ReactMethod
-    public void multiply(int a, int b, Promise promise) {
-        promise.resolve(a * b);
+      promise.resolve(mimeType);
+      return;
+    } catch (Exception e) {
+      promise.reject("Error getting mime type", e);
     }
+  }
 
-    public static native int nativeMultiply(int a, int b);
+  /**
+   * Gets the original date time of the video or image file based on the path passed in. The
+   * ISO datetime is retrieved from the Exif data on the image or video file.
+   *
+   * @param uri       - The video or image file path to get the timestamp of.
+   * @param mediaType - Either 'video' or 'image' so the method knows how to process the media file.
+   * @returns ISO datetime of the image or video file from the file's Exif data.
+   */
+  @ReactMethod
+  public void getTimestamp(String uri, String mediaType, Promise promise) {
+    try {
+      Uri fileUri = Uri.parse(uri);
+
+      // Handle getting mime type for images
+      if (mediaType.equalsIgnoreCase("image")) {
+        InputStream inputStream = getReactApplicationContext().getContentResolver().openInputStream(fileUri);
+        ExifInterface exif = null;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
+          exif = new ExifInterface(inputStream);
+        }
+
+        String timestamp = exif.getAttribute(ExifInterface.TAG_DATETIME);
+        Date date = new SimpleDateFormat("yyyy:MM:dd HH:mm:ss", Locale.US).parse(timestamp);
+        String formattedDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'", Locale.US).format(date);
+
+        promise.resolve(formattedDate);
+
+        // Handle getting mime type for videos
+      } else {
+        MediaMetadataRetriever retriever = GetMediaMetadataRetriever(fileUri);
+        String timestamp = retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DATE);
+        Date date = new SimpleDateFormat("yyyyMMdd'T'HHmmss", Locale.US).parse(timestamp);
+        String formattedDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'", Locale.US).format(date);
+        retriever.release();
+
+        promise.resolve(formattedDate);
+      }
+    } catch (Exception e) {
+      promise.reject("Error getting timestamp of media file", e);
+    }
+  }
+
+  private MediaMetadataRetriever GetMediaMetadataRetriever(Uri fileUri) throws FileNotFoundException {
```

Comment:
Not sure if it's standard to capitalise function names in Java

```suggestion
  private MediaMetadataRetriever getMediaMetadataRetriever(Uri fileUri) throws FileNotFoundException {
```